### PR TITLE
 Fix renaming issues for postIncrement, foreach-loop, object fields

### PR DESCRIPTION
**Pull request by rasika**

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/position/PositionTreeVisitor.java [hunk 1]
```diff
@@ -141,6 +141,20 @@ public void visit(BLangFunction funcNode) {
             return;
         }
 
+        if (HoverUtil.isMatchingPosition(HoverUtil.getIdentifierPosition(funcNode), this.position)) {
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/position/PositionTreeVisitor.java]: Is there any possibility to introduce a common method for the following duplications?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java [hunk 2]
```diff
@@ -168,6 +168,13 @@ public static DiagnosticPos toZeroBasedPosition(DiagnosticPos diagnosticPos) {
         return new DiagnosticPos(diagnosticPos.getSource(), startLine, endLine, startColumn, endColumn);
     }
 
+    public static void replacePosition(DiagnosticPos oldPos, DiagnosticPos newPos) {
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java]: missing doc comment for public method

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/util/HoverUtil.java [hunk 3]
```diff
@@ -346,6 +350,78 @@ public static DiagnosticPos getIdentifierPosition(BLangEndpoint endpointNode) {
...
+            for (Whitespace w : ws) {
+                serviceTypeLength += w.getPrevious().length() + w.getWs().length();
+            }
+            position.sCol += (serviceTypeLength + serviceKeywordLength);
+            position.eCol = position.sCol + serviceNode.name.value.length();
+        }
+        return position;
+    }
+
+    public static DiagnosticPos getIdentifierPosition(BLangResource resource) {
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/util/HoverUtil.java]: missing doc comment for public method

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/position/PositionTreeVisitor.java (hunk 1 above)
Comment (rasika) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/position/PositionTreeVisitor.java]: +1, will move to a common method

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/rename/RenameTest.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        boolean isSubList = true;
+        for (JsonElement exp : subArray) {
+            isSubList = false;
+            for (JsonElement res : array) {
+                if (exp.equals(res)) {
+                    isSubList = true;
+                    break;
+                }
+            }
+            if (!isSubList) {
```
[the remark this answers is not in the document]
Comment (rasika) [reply language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/rename/RenameTest.java]: Yes. It's inevitable IMO. Need to go through all elements in a JsonArray object. First `break` is to break iteration of inner loop, when we found the correct element. Second `break` is to break if atleast one of element is non-existent no point of continuing, just breaks outer loop.

[the remark this answers is not in the document]
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/rename/RenameTest.java]: Great. Let's stick to this :)

**Pull request merged**